Source/Rhetos.Utilities/ApplicationConfiguration/ConfigurationSources/FileSourceConfigurationValue.cs

```diff
@@ -0,0 +1,33 @@
+﻿/*
+    Copyright (C) 2014 Omega software d.o.o.
+
+    This file is part of Rhetos.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+namespace Rhetos.Utilities.ApplicationConfiguration.ConfigurationSources
+{
+    public class FileSourceConfigurationValue : IConfigurationValue
+    {
+        public object Value { get; }
+        public string FilePath { get; }

```

Comment:
I think it would be cleaner to use a folder here instead of the FilePath. File name seens like an excess information that complicates readability and usage. For example, it we want to add a relative path in specific folder, we will have to fake the filename that does not need to exist.

It might be helpful information for debugging, but for that purpose it would be simpler and more complete solution to keep a reference to IConfigurationSource instance.

Reply:
`FilePath` is much more future-proof and incurs very little cost. It is a unique origin of the configuration value (for example 2 configuration files loaded from the same folder). Would definitely keep full file path.

Reply:
Larger refactor of ConfigurationValue and sources.